src/process-cache.js

```diff
@@ -4,32 +4,61 @@ import cwdRelativePath from "./cwd-relative-path.js";
 import ErrorCodes from "./error-codes.js";
 import Format from "./format.js";
 import validateAndFix from "./validate-and-fix.js";
+import validateAndJson from "./validate-and-json.js";
 import validateAndReport from "./validate-and-report.js";
-
 import defaultArgs from "./default-args.js";
+import {version} from "../package.json";
 
 import type {ErrorCode} from "./error-codes.js";
-import type {MarkerCache, ILog, Options} from "./types";
+import type {MarkerCache, ILog, Options, JsonItem} from "./types";
 
 export default async function processCache(
     options: Options,
     cache: $ReadOnly<MarkerCache>,
     log: ILog,
 ): Promise<ErrorCode> {
-    const {autoFix} = options;
+    const {autoFix, json} = options;
     const violationFileNames: Array<string> = [];
-    const fileValidator = autoFix ? validateAndFix : validateAndReport;
-    for (const file of Object.keys(cache)) {
-        try {
-            if (!(await fileValidator(options, file, cache, log))) {
-                violationFileNames.push(file);
+
+    if (json) {
+        const jsonItems: Array<JsonItem> = [];
+        for (const file of Object.keys(cache)) {
+            try {
+                validateAndJson(options, file, cache, jsonItems);
+            } catch (e) {
+                log.error(
+                    `${Format.cwdFilePath(
+                        cwdRelativePath(file),
+                    )} update encountered error: ${e.message}`,
+                );
+            }
+        }
```

Comment:
I think that could work.  I'll need to massage the branch that uses `const fileValidator = autoFix ? validateAndFix : validateAndReport;` a bit, but it shouldn't be too bad.

Reply:
Yeah, it would just mean there was perhaps a `getValidator` call that worked out which one. Like

```
getValidator = (options, blah) => {
   if (autoFix) {
      return validateAndFix;
   }
   
   if (json) {
      return (options, file, cache) => validateAndJson(options, file, cache, jsonItems);
   }
   
   return validateAndReport;
};
```